lib/compilers/typescript-native.ts

```diff
@@ -74,13 +85,22 @@ export class TypeScriptNativeCompiler extends BaseCompiler {
             execOptions,
         );
         if (output.code !== 0) {
-            return [{text: 'Failed to run compiler to get MLIR code'}];
+            return {
+                code: output.code,
+                timedOut: false,
+                stdout: [],
+                stderr: [
+                    {
+                        text: 'Failed to run compiler to get MLIR code',
+                    },
+                ],
+            };
```

Comment:
Not sure if this is correct, made a best effort to get things into the correct form

Reply:
100%